File: ProjectEuler3/main.cpp
#include <iostream>
#include <chrono>
#include <cmath>

int main() {
    std::chrono::time_point<std::chrono::system_clock> start, end;
    start = std::chrono::system_clock::now();

    long number = 600851475143;

    for (int i = 2; i <= sqrt(number); ++i)
    {
        if(number % i == 0) {
            number /= i;
            i--;
        }
    }

    end = std::chrono::system_clock::now();
    long elapsed_millis = std::chrono::duration_cast<std::chrono::milliseconds>(end-start).count();
    std::cout << "Elapsed Milliseconds: " << elapsed_millis << std::endl;

    std::cout << number << std::endl;
    return 0;
}

File: ProjectEuler3/src/main.rs
use std::time::{Duration, Instant};

fn main() {
    let now = Instant::now();

    let mut number:i64 = 600851475143;
    let sqrt_of_number = (number as f64).sqrt() as i64;
    let mut divider = 2;

    while number != 1
    {
        if divider > sqrt_of_number{
            divider = number;
            break;
        } else if number % divider == 0 {
            number /=  divider;
        } else {
            divider += 1;
        }
    }

    println!("Number: {}", divider);
    println!("Elapsed µs: {}", now.elapsed().as_micros());
}
